pandas/core/arrays/integer.py

```diff
@@ -414,6 +415,7 @@ def __setitem__(self, key, value):
             value = value[0]
             mask = mask[0]
 
+        key = check_array_indexer(self, key)
```

Comment:
Can you add the same to BooleanArray and StringArray ?

Reply:
thanks, added!